misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW int_range_view1 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (3, '[2,6)'::INT8RANGE);
+                > UPDATE int_range_table SET f2 = '[2,2)'::INT8RANGE WHERE f1 = 3;
+            """,
+                """
+                > CREATE MATERIALIZED VIEW int_range_view2 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (4, '[2,20]'::INT8RANGE);
+            """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > SELECT * FROM int_range_table ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+                        
+            > SELECT * FROM int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT * FROM int_range_view2 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, COALESCE(f2, '[10,80)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 [10,80)
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, GREATEST(f2, '[2,5]'::INT8RANGE), LEAST(f2, '[2,5]'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) [2,6)
+            2 [2,6) [2,6)
+            3 [2,6) empty
+            4 [2,21) [2,6)
+
+            > SELECT f1, f2, NULLIF(f2, '[2,17)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) <null>
+            2 <null> <null>
+            3 empty empty
+            4 [2,21) [2,21)
+
+            > SELECT f1, f2, LOWER(f2), UPPER(f2), ISEMPTY(f2), LOWER_INC(f2), UPPER_INC(f2), LOWER_INF(f2), UPPER_INF(f2) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) 2 17 false true false false false
+            2 <null> <null> <null> <null> <null> <null> <null> <null>
+            3 empty <null> <null> true false false false false
+            4 [2,21) 2 21 false true false false false
+
+            > SELECT f1, f2, f2 < '[2,16]'::INT8RANGE AS lt, f2 <= '[2,16]'::INT8RANGE AS le, f2 = '[2,16]'::INT8RANGE AS eq, f2 >= '[2,16]'::INT8RANGE AS ge, f2 > '[2,16]'::INT8RANGE AS gt from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) true true true true true
+            2 <null> <null> <null> <null> <null> <null>
+            3 empty false false false false false
+            4 [2,21) false false false false false
+
+            > SELECT f1, f2, f2 + '[3,20]'::INT8RANGE AS merge, f2 * '[8,20]'::INT8RANGE AS intersec, f2 - '[8,20]'::INT8RANGE AS diff from int_range_view1 ORDER BY f1 ASC;
```

Comment:
You are welcome to add a few more operators -- only select ones that only apply to ranges, e.g. no need to have `+` in there.

Reply:
Ok, will. As for `+`, this is not an arithmetic addition but rather the sign for `union` operations when operating on ranges (see https://www.postgresql.org/docs/9.3/functions-range.html).

Reply:
Yes, it is not the same thing -- but given that the primary task of this test is to serialize and deserialize DDL definitions, it would be more important to represent the operators that are , hm, non traditional like `- | -` .

Reply:
Might also be interesting to store the calculation with operator in materialized view. Then it has to be serialized and deserialized.